src/BenchmarkDotNet.Core/Mathematics/MeasurementsStatistics.cs

```diff
@@ -0,0 +1,122 @@
+﻿using BenchmarkDotNet.Reports;
+using System;
+using System.Collections.Generic;
+
+namespace BenchmarkDotNet.Mathematics
+{
+    /// <summary>
+    /// the goal of this struct is to avoid any heap allocations, please keep it in mind
+    /// </summary>
+    internal struct MeasurementsStatistics
+    {
+        public double StandardError { get; }
+        public double Mean { get; }
+
+        private MeasurementsStatistics(double standardError, double mean)
+        {
+            StandardError = standardError;
+            Mean = mean;
+        }
+
+        public static MeasurementsStatistics Calculate(List<Measurement> measurements, bool removeOutliers)
+        {
+            var N = measurements.Count;
+            if (N == 0)
+                throw new InvalidOperationException("StatSummary: Sequence contains no elements");
+
+            var sum = Sum(measurements);
+            var mean = sum / N;
+
+            var variance = Variance(measurements, N, mean);
+            var standardDeviation = Math.Sqrt(variance);
+            var standardError = standardDeviation / Math.Sqrt(N);
+
+            if (!removeOutliers) // most simple scenario is done without allocations! but this is not the default case
+                return new MeasurementsStatistics(standardError, mean);
+
+            measurements.Sort(); // sort in place
+
+            double Q1, Median, Q3;
+
+            if (N == 1)
+                Q1 = Median = Q3 = measurements[0].Nanoseconds;
+            else
+            {
+                Q1 = GetQuartile(measurements, measurements.Count / 2);
+                Median = GetQuartile(measurements, measurements.Count);
+                Q3 = GetQuartile(measurements, measurements.Count * 3 / 2);
+            }
+
+            var InterquartileRange = Q3 - Q1;
+            var LowerFence = Q1 - 1.5 * InterquartileRange;
+            var UpperFence = Q3 + 1.5 * InterquartileRange;
```

Comment:
Could you use lower first letter (because here we are using local variables instead of properties)?